electionleaflets/apps/leaflets/forms.py

```diff
@@ -139,7 +139,10 @@ def get_ballot_data_from_ynr(self, postcode):
         start, end = self.get_date_range()
         params["election_date_range_after"] = start
         params["election_date_range_before"] = end
-        req = requests.get(url, params=params)
+        try:
+            req = requests.get(url, params=params)
+        except requests.RequestException:
+            return []
         req.raise_for_status()
```

Comment:
The thing that will throw an exception is `req.raise_for_status()`. That should be inside your `try:` block

Reply:
also grumble grumble calling the response `req` grumble grumble

Reply:
I went back and forth on including the `raise_for_status` in the try block. I was actually wanting to catch network connection errors (e.g no internet connection on the client) rather than, e.g a 500 on the server. I guess a 503 or some such might be  something we want to catch.

Rather than enumerating these, I've just added it to the block. The worst that happens is that we don't get the actual candidate attached to the leaflet.